python/main.py

```diff
@@ -51,33 +55,97 @@ async def lifespan(app: FastAPI):
     allow_headers=["*"],
 )
 
+class Item(BaseModel):
+    name: str
+    category: str
+    image_name: str
+
 
 class HelloResponse(BaseModel):
     message: str
 
+class AddItemResponse(BaseModel):
+    message: str
+
+class GetItemsResponse(BaseModel):
+    items: List[Item]
+
+class GetItemResponse(BaseModel):
+    item: Item
+
 
 @app.get("/", response_model=HelloResponse)
 def hello():
     return HelloResponse(**{"message": "Hello, world!"})
 
 
-class AddItemResponse(BaseModel):
-    message: str
-
-
 # add_item is a handler to add a new item for POST /items .
 @app.post("/items", response_model=AddItemResponse)
-def add_item(
+async def add_item(
     name: str = Form(...),
+    category: str = Form(...),
+    image: UploadFile = File(...),
     db: sqlite3.Connection = Depends(get_db),
 ):
     if not name:
         raise HTTPException(status_code=400, detail="name is required")
-
-    insert_item(Item(name=name))
+    if not category:
+        raise HTTPException(status_code=400, detail="category is required")
+
+    #load image
+    image_contents = await image.read() 
+
+    # Hashing images with SHA-256
+    sha256 = hashlib.sha256(image_contents).hexdigest()
+    image_name = f"{sha256}.jpg"  
+
+    # Save images to images directory
+    image_path = images / image_name
+    with open(image_path, "wb") as f:
+        f.write(image_contents)
+    
+    # insert items to items.json
+    insert_item(Item(name=name, category=category, image_name=image_name))
+    
     return AddItemResponse(**{"message": f"item received: {name}"})
 
 
+# STEP 4-3: Retrieve product list
+@app.get("/items", response_model=GetItemsResponse)
+def get_items():
+    # if no json data
+    if not ITEMS_FILE_PATH.exists():
+        return GetItemsResponse(items=[])
+    
+    with open(ITEMS_FILE_PATH, "r") as f:
+        data = json.load(f)
+    
+    items = []
+    for item in data["items"]:
+        items.append(item)
+    
+    return GetItemsResponse(items=items)
+
+#STEP 4-5: Return product details
+@app.get("/items/{item_id}", response_model=GetItemResponse)
+def get_item(item_id: int):
+    # if no json data
+    if not ITEMS_FILE_PATH.exists():
+        raise HTTPException(status_code=404, detail="item is not found")
+    
+    with open(ITEMS_FILE_PATH, "r") as f:
+        data = json.load(f)
+
+    # check if item_id exists
+    if item_id > len(data["items"]) and item_id > 0:
```

Comment:
I think this is a wrong condition, when `item_id` is 0 or negative, it won't throw the error & code would break at line 144

The correct condition will be

```suggestion
    if item_id <= 0 or item_id > len(data["items"]) :
```